### PR TITLE
GEODE-2484: Replaces ACE with boost::filesystem in stats.

### PR DESCRIPTION
**Pull request by jake-at-work**

### CODE REVIEW

File: cppcache/src/SystemProperties.cpp [hunk 1]
```diff
@@ -88,7 +88,7 @@ constexpr auto DefaultConnectWaitTimeout = std::chrono::seconds::zero();
 constexpr auto DefaultBucketWaitTimeout = std::chrono::seconds::zero();
 
 constexpr auto DefaultSamplingInterval = std::chrono::seconds(1);
-const bool DefaultSamplingEnabled = true;
+constexpr bool DefaultSamplingEnabled = false;
```
Comment (pdxcodemonkey) [inline cppcache/src/SystemProperties.cpp:91]: auto?

**Review by pdxcodemonkey — CHANGES_REQUESTED** [review]

File: cppcache/include/geode/CacheableString.hpp [hunk 2]
```diff
@@ -25,8 +25,7 @@
 #include "internal/DataSerializablePrimitive.hpp"
 #include "internal/geode_globals.hpp"
 
-/** @file
- */
+#undef max
```
[the remark this answers is not in the document]
Comment (pivotal-jbarrett) [reply cppcache/include/geode/CacheableString.hpp:28]: C preprocessor rules suck. Something included above results in leaking a macro from a system header defined as `max`. This then collides with `std::max()` resulting gibberish that won't compile. I think I may have already corrected the issue by cleaning up headers. I will pull this and see what happens now.

File: cppcache/src/AdminRegion.hpp-a968f121 [hunk 3]
```diff
@@ -0,0 +1,84 @@
+/*
```
[the remark this answers is not in the document]
Comment (pivotal-jbarrett) [reply cppcache/src/AdminRegion.hpp-a968f121:1]: 🤦 Whoops!

File: cppcache/src/CppCacheLibrary.hpp [hunk 4]
```diff
@@ -31,17 +31,28 @@ namespace client {
 // initialize GEODE runtime if it has not already been initialized.
 class APACHE_GEODE_EXPORT CppCacheLibrary {
  public:
-  // Call to this to trigger initialization.
+  /**
+   * Call to this to trigger initialization.
+   */
   static void initLib(void);
```
[the remark this answers is not in the document]
Comment (pivotal-jbarrett) [reply cppcache/src/CppCacheLibrary.hpp:37]: This I didn't want to mess with! Many versions and compilers ago I tried and the ordering changed and resulted in bad things. I don't even want to look at this wrong.

[the remark this answers is not in the document]
Comment (pdxcodemonkey) [reply cppcache/src/CppCacheLibrary.hpp:37]: https://issues.apache.org/jira/browse/GEODE-8208

**Review by pdxcodemonkey — APPROVED** [review]

**Pull request merged**